### ISSUE TITLE
Smileys don't appear (Windows 2012R2 IIS 8.5)

### ISSUE THREAD

**Issue by furyritchie** [iionly/elggchat#2]
The chat seems to work well but smileys don't seem to work as originally planned. I'm running IIS which is unsupported by Elgg but everything seems to work as expected. See the picture to see the two things I usually see instead of the smileys.
![9gum6](https://cloud.githubusercontent.com/assets/12377889/12533017/fd260ae8-c276-11e5-9d31-264dd4f45eb0.png)

**Comment by iionly** [comment]
I'm afraid I can't give much useful advice here, because I've absolutely no experience with Windows servers.

The "X" might be some indication that the corresponding smiley image can't be found. Or the access permissions are set wrongly, i.e. the server just can't read the file. But I don't know what the permissions would have to be set to on a Windows server - maybe it helps to set them simply to the same permissions then other similar files / the other files of the Elggchat plugin.

But the faulty display of the smilies in the second seems to indicate that there might be some other issue causing the problem. I really can't say what it is though as I've never seen something similar before. Maybe it's not even a server issue but rather a browser issue.

Maybe it helps to check the server log for error entries that might be connected with the smilies. Maybe some general issues with Elgg on the server are present. But I've no idea what it might be.

**Comment by furyritchie** [comment]
Thanks for your quick reply.

I've checked the permissions and they're the same as everything else. 

Those unusual images appear in Microsoft Edge. In Google Chrome I get something strange showing up before it just disappears and it's blank where the smiley should be (whitespace).

I will check the log at a later stage. I might move this back to a linux server instead.

**Comment by iionly** [comment]
Possibly the problem with the incorrect display of the smilies is connected with the header content sent together with the smiley images. Currently, no mime-type is sent by the Elggchat plugin. It could be that some browsers (possibly even different with the same browser on different OSes) are handling this more strict. I hadn't seen any issues during testing so far and therefore didn't expect any issues. Though I had seen some issues with images not correctly displayed with some other plugins and these issues could be fixed by changes in the header. I'll look into it (working on Elggchat anyway soon for some small issues). Still, I have no explanation for the effect with the smilies in the second example message. But maybe it's the same issue nevertheless.

**Comment by iionly** [comment]
I've made new releases for Elgg 1.10-1.12 (version 1.10.7) and for Elgg 2.X (version 2.0.1) where I added the mime-type to the smiley images server. Hopefully, this solves your issue on the Windows server (or maybe rather within the browser on the Windows client). I don't think I can do anything else (as nothing is broken for me in the first place and it's kind of impossible to fix without being able to reproduce).
